### DISCUSSION TITLE
Text input with rotary encoder

### DISCUSSION THREAD

**Discussion by tomelgato** [davetcc/tcMenu#461] [General]
Before i reinvent the wheel, is there a library for text input with a rotary encoder/lcd?

**Comment by davetcc** [comment]
If you mean in terms of TcMenu, it can support LCD displays and text input using TextMenuItem.

If you mean standalone, no idea.

**Comment by tomelgato** [comment]
Cool, i did miss that cause i didnt use an encoder before. So TextMenuItem allows me to enter/edit a name/text just with the rotary encoder and encoder pushbutton?

**Reply by davetcc** [reply]
Yes, it's got multi-edit capability for any input type, you can even write your own custom multi-editors easily too:

https://www.thecoderscorner.com/products/arduino-libraries/tc-menu/menu-item-types/editabletext-menu-item/
https://www.thecoderscorner.com/products/arduino-libraries/tc-menu/menu-item-types/rgb32-menu-item/

**Reply by davetcc** [reply]
I've noticed that most platformIo users are not aware of the vast range of examples that are shipped with the library, covering almost all capabilities. I recommend that you take a copy of the library locally just to get the examples.
